src/platform/packages/shared/kbn-esql-validation-autocomplete/src/definitions/types.ts

```diff
@@ -372,6 +378,17 @@ export interface CommandSuggestParams<CommandName extends string> {
     queryString: string,
     prefix?: string
   ) => Promise<SuggestionRawDefinition[]>;
+
+  /**
+   * Returns a list of recommended fields or functions based on the type.
+   * @param type - The type to suggest fields or functions for.
+   * @returns
+   */
```

Comment:
This does a bit more than this, right? You're examining the command and filtering out columns that are already used?

The fate of `getUserDefinedColumns` is very uncertain since it goes against our strategy of making things command-specific. No strong opinion, but I don't know how much we want to build on top of it at this point.